server/config.go

```diff
@@ -161,6 +165,17 @@ func (cfg *Config) VerificationCfg() *verify.Config {
 
 // ReadConfig ... parses the Config from the provided flags or environment variables.
 func ReadConfig(ctx *cli.Context) Config {
+	signerPrivateKeyHex := func() string {
+		if key := ctx.String(SignerPrivateKeyHexFlagName); key != "" {
+			return key
+		}
+		randomBytes := make([]byte, 32)
+		if _, err := rand.Read(randomBytes); err != nil {
+			return ""
+		}
+		return hex.EncodeToString(randomBytes)
+	}()
```

Comment:
how do we handle the case in-which read mode is disable and the user didn't provide a valid key? should we fail loudly assuming this could be an error case that someone running in production would need to know about?

Reply:
knit - would prefer this to live as a separate function within this file to avoid overlaying additional programming abstractions (i.e, callbacks) that aren't generally adopted within the service